frame/treasury/src/lib.rs

```diff
@@ -198,6 +220,32 @@ pub struct OpenTip<
 	tips: Vec<(AccountId, Balance)>,
 }
 
+/// An index of a bounty. Just a `u32`.
+pub type BountyIndex = u32;
+
+/// A bounty proposal.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub struct Bounty<AccountId, Balance> {
+	/// The account proposing it.
+	proposer: AccountId,
+	/// The account manages this bounty.
+	curator: AccountId,
+	/// The (total) amount that should be paid if the bounty is rewarded.
+	value: Balance,
+	/// The amount held on deposit (reserved) for making this proposal.
+	bond: Balance,
+	/// The description of this bounty.
+	description: Vec<u8>,
```

Comment:
May be better as a `HashOrData` to constrain size on-chain?

Reply:
The `reason` of `report_awesome` is also just `Vec<u8>` and stored on-chain.
Maybe we want to move `Data` from pallet-identity to frame-support and reuse it?

https://github.com/paritytech/substrate/blob/812d94d42c6d6d7fd28cf18a08d84f14c49d8544/frame/identity/src/lib.rs#L127-L149